src/snovault/viewconfigs/searchview.py

```diff
@@ -0,0 +1,320 @@
+    sort_query,
+    get_filtered_query,
+    set_sort_order,
+    get_search_fields,
+    list_visible_columns_for_schemas,
+    list_result_fields,
+    set_filters,
+    set_facets,
+    format_facets,
+    iter_long_json,
+    format_results,
+    search_result_actions
+)
+
+
+DEFAULT_DOC_TYPES = [
+    'AntibodyLot',
+    'Award',
+    'Biosample',
+    'Dataset',
+    'GeneticModification',
+    'Page',
+    'Pipeline',
+    'Publication',
+    'Software',
+    'Target',
+]
+
+
+audit_facets = [
+    ('audit.ERROR.category', {'title': 'Audit category: ERROR'}),
+    ('audit.NOT_COMPLIANT.category', {'title': 'Audit category: NOT COMPLIANT'}),
+    ('audit.WARNING.category', {'title': 'Audit category: WARNING'}),
+    ('audit.INTERNAL_ACTION.category', {'title': 'Audit category: DCC ACTION'})
+]
+
+
+class SearchView(BaseView):
+    def __init__(self, context, request, search_type=None, return_generator=False):
+        super(SearchView, self).__init__(context, request)
+        self.search_type = search_type
+        self.return_generator = return_generator
+
+    def construct_clear_filters(self):
+        searchterm_specs = self.request.params.getall('searchTerm')
+        searchterm_only = urlencode([("searchTerm", searchterm) for searchterm in searchterm_specs])
+        if searchterm_only:
+            # Search term in query string; clearing keeps that
+            clear_qs = searchterm_only
+        else:
+            # Possibly type(s) in query string
+            clear_qs = urlencode([("type", typ) for typ in self.doc_types])
+        return self.request.route_path('search', slash='/') + (('?' + clear_qs) if clear_qs else '')
+
+    def construct_filters(self):
+
+        for item_type in self.doc_types:
+            ti = self.types[item_type]
+            qs = urlencode([
+                (k.encode('utf-8'), v.encode('utf-8'))
+                for k, v in self.request.params.items() if not (k == 'type' and self.types['Item' if v == '*' else v] is ti)
+            ])
+            filters = {
+                'field': 'type',
+                'term': ti.name,
+                'remove': '{}?{}'.format(self.request.path, qs)
+            }
+        return filters
+
+    def construct_doc_types(self, doc_types=None):
+        if doc_types is not None:
+            self.doc_types = doc_types
+            return
+        # Process doc_types
+        if self.search_type is None:
+            if '*' in self.doc_types:
+                self.doc_types = ['Item']
+
+        else:
+            self.doc_types = [self.search_type]
+
+        try:
+            self.doc_types = sorted({self.types[name].name for name in self.doc_types})
+        except KeyError:
+            # Check for invalid types
+            bad_types = [t for t in self.doc_types if t not in self.types]
+            msg = "Invalid type: {}".format(', '.join(bad_types))
+            raise HTTPBadRequest(explanation=msg)
+
+        if not self.doc_types:
```

Comment:
See other comment in preprocess.  This conditional needs to happen after construct_clear_filters.  And the code above this line in construct_doc_types needs to happen before construct_clear_filters